src/repos.cc

```diff
@@ -2096,23 +2129,8 @@ void modify_repo(Zypper & zypper, const string & alias)
       repo.setGpgCheck(gpgCheck);
     }
 
-    long long prio = 0;
-    parsed_opts::const_iterator tmp1;
-    if ((tmp1 = zypper.cOpts().find("priority")) != zypper.cOpts().end())
-    {
-      //! \todo use some preset priorities (high, medium, low, ...)
-
-      string prio_str = *tmp1->second.begin();
-      safe_lexical_cast(prio_str, prio); // try to make an int out of the string
-      if (prio < 1)
-      {
-        zypper.out().error(boost::str(format(
-          _("Invalid priority '%s'. Use a positive integer number. The greater the number, the lower the priority."))
-          % prio_str));
-        zypper.setExitCode(ZYPPER_EXIT_ERR_INVALID_ARGS);
-        return;
-      }
-
+    long long prio = parse_priority(zypper);
```

Comment:
```unsigned...``` and the parser call should be moved up, outside the ```try```-block. No need to setup anything if the cli parsing fails.